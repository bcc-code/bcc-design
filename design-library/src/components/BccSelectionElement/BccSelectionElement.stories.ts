import BccSelectionElement from "./BccSelectionElement.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

export default {
  title: "Components/BccSelectionElement",
  component: BccSelectionElement,
  argTypes: {},
} as Meta<typeof BccSelectionElement>;

const Template: StoryFn<typeof BccSelectionElement> = (args) => ({
  components: { BccSelectionElement },
  setup() {
    return { args };
  },
  template: `
    <BccSelectionElement>Link to first page</BccSelectionElement>
    <BccSelectionElement>Link to second page</BccSelectionElement>
    <BccSelectionElement :isLast="args.isLast ?? true">Link to last page</BccSelectionElement>
  `,
});

export const Example = Template.bind({});
Example.args = {};
