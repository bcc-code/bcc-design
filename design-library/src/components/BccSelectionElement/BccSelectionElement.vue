<script setup lang="ts">
import { ArrowForwardIcon } from "@bcc-code/icons-vue";
type Props = {
  isLast?: boolean;
};

withDefaults(defineProps<Props>(), {});
</script>

<template>
  <div
    class="border-secondary group flex w-full cursor-pointer justify-between border-b py-2"
    :class="{ 'border-none': isLast }"
  >
    <label class="text-label font-bold"><slot></slot></label>
    <div class="pr-1.5 transition-all group-hover:pr-0">
      <ArrowForwardIcon class="h-4 w-4" />
    </div>
  </div>
</template>
