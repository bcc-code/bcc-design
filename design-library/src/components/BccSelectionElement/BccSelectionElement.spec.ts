import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccSelectionElement from "./BccSelectionElement.vue";

describe("BccSelectionElement", () => {
  it("renders", async () => {
    const wrapper = mount(BccSelectionElement, { props: {} });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders last element", async () => {
    const wrapper = mount(BccSelectionElement, { props: { isLast: true } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
