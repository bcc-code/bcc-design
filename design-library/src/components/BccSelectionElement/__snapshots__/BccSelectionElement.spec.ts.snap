// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BccSelectionElement > renders 1`] = `
"<div class=\\"border-secondary group flex w-full cursor-pointer justify-between border-b py-2\\"><label class=\\"text-label cursor-pointer font-bold\\"></label>
  <div class=\\"pr-1.5 transition-all group-hover:pr-0\\"><svg aria-hidden=\\"true\\" viewBox=\\"0 -960 960 960\\" fill=\\"currentColor\\" class=\\"h-4 w-4\\">
      <path d=\\"M647-440H200q-17 0-28.5-11.5T160-480q0-17 11.5-28.5T200-520h447L451-716q-12-12-11.5-28t12.5-28q12-11 28-11.5t28 11.5l264 264q6 6 8.5 13t2.5 15q0 8-2.5 15t-8.5 13L508-188q-11 11-27.5 11T452-188q-12-12-12-28.5t12-28.5l195-195Z\\"></path>
    </svg></div>
</div>"
`;

exports[`BccSelectionElement > renders last element 1`] = `
"<div class=\\"border-secondary group flex w-full cursor-pointer justify-between border-b py-2 border-none\\"><label class=\\"text-label cursor-pointer font-bold\\"></label>
  <div class=\\"pr-1.5 transition-all group-hover:pr-0\\"><svg aria-hidden=\\"true\\" viewBox=\\"0 -960 960 960\\" fill=\\"currentColor\\" class=\\"h-4 w-4\\">
      <path d=\\"M647-440H200q-17 0-28.5-11.5T160-480q0-17 11.5-28.5T200-520h447L451-716q-12-12-11.5-28t12.5-28q12-11 28-11.5t28 11.5l264 264q6 6 8.5 13t2.5 15q0 8-2.5 15t-8.5 13L508-188q-11 11-27.5 11T452-188q-12-12-12-28.5t12-28.5l195-195Z\\"></path>
    </svg></div>
</div>"
`;
